unit-tests/live/calib/test_calibrations_common.py

```diff
@@ -0,0 +1,123 @@
+# This "Software" is furnished under license and may only be used or copied in accordance with the terms of that license.
+# No license, express or implied, by estoppel or otherwise, to any intellectual property rights is granted by this document.
+# The Software is subject to change without notice and should not be construed as a commitment by RealSense, Inc. or Intel Corporation to market, license, sell or support any product or technology.
+# Unless otherwise provided for in the license under which this Software is provided, the Software is provided AS IS, with no warranties of any kind, express or implied.
+# Except as expressly permitted by the Software license, neither RealSense, Inc. nor Intel Corporation, or any of their suppliers, assumes any responsibility or liability for any errors or inaccuracies that may appear herein.
+# Except as expressly permitted by the Software license, no part of the Software may be reproduced, stored in a retrieval system, transmitted in any form, or distributed by any means without the express written consent of RealSense, Inc.
+
+#########################################################################################################################################
+##                   tests for OCC and Tare calib flows                                                                ##
+#########################################################################################################################################
+import sys
+import time
+import pyrealsense2 as rs
+from rspy import test, log
+
+# Constants for calibration
+CALIBRATION_TIMEOUT_SECONDS = 30
+OCC_TIMEOUT_MS = 9000
+TARE_TIMEOUT_MS = 10000
+FRAME_PROCESSING_TIMEOUT_MS = 5000
+HARDWARE_RESET_DELAY_SECONDS = 3
+
+def on_calib_cb(progress):
+    """Callback function for calibration progress reporting."""
+    pp = int(progress)
+    log.i('\r' + '*' * pp + ' ' * (99 - pp) + '*')
+    if pp == 100:
+        log.i("progress - ", pp)
+
+def calibration_main(host_assistance, occ_calib, json_config, ground_truth):
+    """
+    Main calibration function for both OCC and Tare calibrations.
+    
+    Args:
+        host_assistance (bool): Whether to use host assistance mode
+        occ_calib (bool): True for OCC calibration, False for Tare calibration
+        json_config (str): JSON configuration string
+        ground_truth (float): Ground truth value for Tare calibration (None for OCC)
+    
+    Returns:
+        bool: True if calibration was not skipped, False if was skipped
+        float: Health factor from calibration
+    """
+    config = rs.config()
+    pipeline = rs.pipeline()
+    pipeline_wrapper = rs.pipeline_wrapper(pipeline)
+    pipeline_profile = config.resolve(pipeline_wrapper)
+    auto_calibrated_device = rs.auto_calibrated_device(pipeline_profile.get_device())
+    if not auto_calibrated_device:
+        log.e("Failed to open device for calibration")
```

Comment:
Instead of logging and failing test raise an exception and let actual test decide if it's failure and how to log it.